client/src/www/app/method_channel.ts

```diff
@@ -12,14 +12,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {pluginExecWithErrorCode} from './plugin.cordova';
-import {ResourceFetcher} from './resource_fetcher';
+export interface MethodChannel {
+  invokeMethod(methodName: string, params: string): Promise<string>;
+}
+
+let defaultMethodChannel: MethodChannel;
+
+export function installDefaultMethodChannel(
+  methodChannel: MethodChannel
+): void {
+  defaultMethodChannel = methodChannel;
+}
 
-/**
- * Fetches resources using Cordova plugin.
- */
-export class CordovaResourceFetcher implements ResourceFetcher {
-  fetch(url: string): Promise<string> {
-    return pluginExecWithErrorCode<string>('fetchResource', url);
-  }
+export function getDefaultMethodChannel(): MethodChannel {
+  return defaultMethodChannel;
```

Comment:
*nit* Would it be better to check for `null` and `throw Error` when no `defaultMethodChannel` has been installed?

Reply:
Good idea. Done.